Formula/mr2.rb

```diff
@@ -0,0 +1,20 @@
+class Mr2 < Formula
+  desc "expose local server to external network"
+  homepage "https://github.com/txthinking/mr2"
+  url "https://github.com/txthinking/mr2/archive/refs/tags/v20210401.tar.gz"
+  sha256 "3cf2874a5945e79fd9ca270181de1a9d6a662434455c58e2e20e5dbfdebd64c7"
+  license "GPL-3.0-only"
+
+  depends_on "go" => :build
+
+  def install
+    ldflags = "-s -w"
+    cd "cli/mr2" do
+      system "go", "build", *std_go_args, "-ldflags", ldflags
+    end
+  end
+
+  test do
+    assert_match shell_output("#{bin}/mr2"), shell_output("#{bin}/mr2 -h")
```

Comment:
If both commands fail with the same error this would pass.

In addition to that, we need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.